envoy/config/filter/http/jwt_authn/v2/config.proto

```diff
@@ -0,0 +1,207 @@
+
+syntax = "proto3";
+
+package envoy.config.filter.http.jwtauthn.v2;
+
+import "envoy/api/v2/core/http_uri.proto";
+import "envoy/api/v2/route/route.proto";
+import "google/protobuf/duration.proto";
+import "validate/validate.proto";
+
+// JSON Web Token (JWT) token format for authentication is defined
+// `here <https://tools.ietf.org/html/rfc7519>`_.
+// See `OAuth2.0 <https://tools.ietf.org/html/rfc6749>`_ and
+// `OIDC1.0 <http://openid.net/connect>`_ for the authentication flow.
+//
+// Example:
+//
+// .. code-block:: yaml
+//
+//     issuer: https://example.com
+//     audiences:
+//     - bookstore_android.apps.googleusercontent.com
+//       bookstore_web.apps.googleusercontent.com
+//     jwks_uri: https://example.com/.well-known/jwks.json
+//
+// [#not-implemented-hide:]
+message JWT {
+  // Identifies the principal that issued the JWT. See
+  // `here <https://tools.ietf.org/html/rfc7519#section-4.1.1>`_.
+  // Usually a URL or an email address.
+  //
+  // Example: https://securetoken.google.com
+  // Example: 1234567-compute@developer.gserviceaccount.com
+  //
+  string issuer = 1;
+
+  // The list of JWT `audiences
+  // <https://tools.ietf.org/html/rfc7519#section-4.1.3>`_. that are allowed to
+  // access. A JWT containing any of these audiences will be accepted.
+  //
+  // Example:
+  //
+  // .. code-block:: yaml
+  //
+  //     audiences:
+  //     - bookstore_android.apps.googleusercontent.com
+  //       bookstore_web.apps.googleusercontent.com
+  //
+  repeated string audiences = 2;
+
+  // `Json web key set <https://tools.ietf.org/html/rfc7517#appendix-A>`_ URI.
+  // Json web key set is fetched from the URI to validate signature of the JWT.
```

Comment:
Not to be too pedantic, but I generally advocate for `JsonWebKey` and `Json` capitalization in message names (for consistency with elsewhere) but for JSON and JSON Web Key when writing this out in comments/docs.

Reply:
Done